src/backend/joanie/tests/core/test_api_enrollment.py

```diff
@@ -609,13 +762,91 @@ def test_api_enrollment_create_has_read_only_fields(self, mock_set):
+                    ),
+                    "enrollment_end": course_run.enrollment_end.isoformat().replace(
+                        "+00:00", "Z"
+                    ),
+                    "start": course_run.start.isoformat().replace("+00:00", "Z"),
+                    "end": course_run.end.isoformat().replace("+00:00", "Z"),
+                    "state": {
+                        "priority": course_run.state["priority"],
+                        "text": course_run.state["text"],
+                        "call_to_action": course_run.state["call_to_action"],
+                        "datetime": course_run.state["datetime"]
+                        .isoformat()
+                        .replace("+00:00", "Z"),
+                    },
+                },
                 "is_active": is_active,
                 "state": "set",
             },
         )
 
+    def test_api_enrollment_create_for_closed_course_run(self):
+        """An authenticated user should not be allowed to enroll to a closed course run."""
+        user = factories.UserFactory()
+        token = self.get_user_token(username=user.username)
+        target_course = factories.CourseFactory()
+        course_run = self.create_closed_course_run(
+            course=target_course,
+            resource_link="http://openedx.test/courses/course-v1:edx+000001+Demo_Course/course",
+        )
+
+        data = {"course_run": course_run.resource_link, "is_active": True}
+
+        response = self.client.post(
+            "/api/enrollments/",
+            data=data,
+            content_type="application/json",
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
+        self.assertEqual(response.status_code, 400)
+        content = json.loads(response.content)
+
+        self.assertEqual(
+            content,
+            {
+                "__all__": [
+                    "You are not allowed to enroll to a course run not opened for enrollment."
+                ]
+            },
+        )
+
+    def test_api_enrollment_create_with_unknown_course_run(self):
+        """An authenticated user should not be allowed to enroll to an unknown course run."""
+
+        user = factories.UserFactory()
+        token = self.get_user_token(username=user.username)
+        data = {"course_run": "this-course-run-does-not-exist", "is_active": True}
+
+        response = self.client.post(
+            "/api/enrollments/",
+            data=data,
+            content_type="application/json",
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
+
+        self.assertEqual(response.status_code, 400)
+        content = json.loads(response.content)
+
+        self.assertEqual(
+            content,
+            {
+                "__all__": [
+                    (
+                        "A course run with resource link "
+                        '"this-course-run-does-not-exist" does not exist.'
+                    )
+                ]
+            },
+        )
+
```

Comment:
```suggestion
        self.assertEqual(response.status_code, 400)
        
        self.assertEqual(
            response.json(),
            {
                "__all__": [
                    (
                        "A course run with resource link "
                        '"this-course-run-does-not-exist" does not exist.'
                    )
                ]
            },
        )

```

Reply:
Oh thanks for the tips ! I always use `response.content` everywhere